### ISSUE TITLE
Massive refactor and things

### ISSUE THREAD

**Issue by DorianGray** [bigplum/lua-resty-mongol#9]
Hey bigplum,

tl;dr :: I'm refactoring this entire library to look more like lua and less like C. Also adding a rockspec for this and the resty bindings library. Should I make a PR or just create a couple new repos?(EG:: are you still interested in maintaining this)

So I am writing an event based web framework that's quite different from anything in any language for ngx_lua. It's called lusty https://github.com/Olivine-Labs/lusty and consists of many tiny modules. I'm currently writing a RESTful storage adapter for mongo(IE::fire put event, old document gets deleted and new gets inserted into configured collection, multiple adapters allowed for each model type so you can store data in multiple database types easily using predicates) and the interface for this library is wonky and the code is messy and I'm OCD about those things.

Basically I'm looking to rewrite this codebase and break every api it has. I'm also talking to daurnimator about perhaps merging this fork upstream and doing a pull request to his repo. My goals here are to unify the disparate repositories for this project and hopefully get one really nice version out there for everyone using openresty/ngx_lua to enjoy.

Thanks,
DorianGray
